authentication/src/main/java/io/camunda/authentication/filters/AdminUserCheckFilter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.filters;
+
+import io.camunda.search.query.RoleQuery;
+import io.camunda.security.configuration.SecurityConfiguration;
+import io.camunda.service.RoleServices;
+import io.camunda.zeebe.protocol.record.value.DefaultRole;
+import io.camunda.zeebe.protocol.record.value.EntityType;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+public class AdminUserCheckFilter extends OncePerRequestFilter {
+
+  public static final String ADMIN_ROLE_ID = DefaultRole.ADMIN.getId();
+  public static final String USER_MEMBERS = "users";
+  private static final Logger LOG = LoggerFactory.getLogger(AdminUserCheckFilter.class);
+  private final SecurityConfiguration securityConfig;
+  private final RoleServices roleServices;
+
+  public AdminUserCheckFilter(
+      final SecurityConfiguration securityConfig, final RoleServices roleServices) {
+    this.securityConfig = securityConfig;
+    this.roleServices = roleServices;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      final HttpServletRequest request,
+      final HttpServletResponse response,
+      final FilterChain filterChain)
+      throws ServletException, IOException {
+    if (request.getRequestURI().endsWith("/identity/setup")) {
+      filterChain.doFilter(request, response);
+      return;
+    }
+
+    final var hasConfiguredAdminUser =
+        !securityConfig
+            .getInitialization()
+            .getDefaultRoles()
+            .getOrDefault(ADMIN_ROLE_ID, Map.of())
+            .getOrDefault(USER_MEMBERS, Set.of())
+            .isEmpty();
+
+    if (hasConfiguredAdminUser) {
+      LOG.debug("Admin user has been configured.");
```

Comment:
❓ will this log with every request coming in (if debug log level is set?)

Reply:
Ah good point, I have to remove this. I only want to log when we actually redirect.